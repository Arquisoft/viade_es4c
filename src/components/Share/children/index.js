import ShareFormComponent from "./share-form.component";

export {ShareFormComponent};