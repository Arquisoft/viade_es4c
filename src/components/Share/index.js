import ShareComponent from "./share.container";

export default ShareComponent;