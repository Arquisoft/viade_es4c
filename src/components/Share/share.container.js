import React, {useState, useCallback} from "react";
import {ShareFormComponent} from "./children";
import {useNotification} from "@inrupt/solid-react-components";

const ShareComponent = (props) => {
	const [friend, setFriend] = useState("");
	const [route, setRoute] = useState("");
	const {webId} = props;
	const {createNotification} = useNotification(webId);

	const sendNotification = useCallback(
		async (content, to, type, license) => {
			try {
				await createNotification(content, to, type, license);
			} catch (error) {
				alert("Error: ShareComponent > sendNotification");
			}
		},
		[createNotification]
	);

	return (
		<div>
			<ShareFormComponent {...{
				webId,
				friend,
				setFriend,
				route,
				setRoute,
				sendNotification
			}} />
		</div>
	);

};

export default ShareComponent;
