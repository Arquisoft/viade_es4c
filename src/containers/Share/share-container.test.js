import React from 'react';
import { shallow } from 'enzyme';
import Share from "./share.container";

describe('Share', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Share debug />);
  
    expect(component).toMatchSnapshot();
  });
});
