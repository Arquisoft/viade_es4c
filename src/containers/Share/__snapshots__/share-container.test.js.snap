// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Share should render correctly in "debug" mode 1`] = `
<div>
  <p>
    .
  </p>
  <p>
    .
  </p>
  <h1>
    Share Routes
  </h1>
  <ShareFormComponent
    friend=""
    route=""
    sendNotification={[Function]}
    setFriend={[Function]}
    setRoute={[Function]}
  />
</div>
`;
